src/aleph_client/asynchronous.py

```diff
@@ -610,6 +612,33 @@ async def get_messages(
         return MessagesResponse(**messages_json)
 
 
+GenericMessage = TypeVar("GenericMessage", bound=AlephMessage)
+
+
+async def get_message(
+    item_hash: str,
+    message_type: Type[GenericMessage] = AlephMessage,
+    channel: Optional[str] = None,
+    session: Optional[ClientSession] = None,
+    api_server: str = settings.API_HOST,
+) -> GenericMessage:
+    """Get a single message from its `item_hash`."""
+    raw_program_message = await get_messages(
+        hashes=[item_hash],
+        session=session,
+        channels=[channel] if channel else None,
+        api_server=api_server,
+    )
+    if len(raw_program_message.messages) < 1:
+        raise MessageNotFoundError(f"No such hash {item_hash}")
+    assert (
```

Comment:
`assert` can be deactivated, I see your point but a proper exception would be better IMO.

```python
messages = raw_program_message.messages
if not messages:
  raise MessageNotFoundError(f"No such hash {item_hash}")
elif len(messages > 2):
  raise TooManyResults("API returned more than one message, which is unexpected.")
```

Reply:
Changed.